### PR TITLE
[ci] Add GitHub actions to lint/test

### PR DESCRIPTION
**Pull request by mpariente**
- Linting with `pylint` and `black`. We can talk about what exceptions we tolerate. Do we want to lint `scripts` as well? 
- For the unit test config, it's not fully tested but it should work. 
- I'm stuck with the requirements, how should we proceed? `environement.yml` is nice, I like it. But having a `requirements.txt` can be useful as well for non-conda users. What do you think about having `requirements.txt` and `dev-requirements.txt` and having them called in the `environement.yml` (with `-r`)? That would make a nice compromise IMO. 
- Codecov should be set up for asteroid but also not sure yet.

### CODE REVIEW

**Comment by mpariente** [comment]
Do you know how conflicts between conda and pip are handled in the env file? e.g if pytorch is both in `dependencies` and `pip`? I guess that pip will just saw requirement satisfied but I'm not sure. In this case, it's fine to repeat numpy/pytorch/scipy in both deps and pip. WDYT?

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[flake8]
+max-line-length = 119
```
Comment (iver56) [inline .flake8:2]: Maybe we should align this value with `line-length` in pyproject.toml? Or is this value higher on purpose to give some leeway?

**Comment by iver56** [comment]
> 
> 
> Do you know how conflicts between conda and pip are handled in the env file? e.g if pytorch is both in `dependencies` and `pip`? I guess that pip will just saw requirement satisfied but I'm not sure. In this case, it's fine to repeat numpy/pytorch/scipy in both deps and pip. WDYT?

I've seen errors where the culprit was that numpy was somehow installed twice (probably by both conda and pip). Haven't had any other negative experiences with combining conda and pip though. Is it troublesome to use conda in Github Actions?

File: .flake8 (hunk 1 above)
Comment (mpariente) [reply .flake8:2]: We can. We didn't agree on the line length yet, so I didn't pay attention to it.

**Comment by mpariente** [comment]
My guess is that setting up conda will take quite some time. 
Also, how does it behave to install cudatoolkit on a CPU-only machine?

**Comment by mpariente** [comment]
Maybe I'm missing something. How would you handle the deps specification between 
- conda env file
- requirements.txt
- setup.py
In particular, the versions are currently fixed in the conda env file (`==`), is this necessary? Would you also fix it in the setup.py?

**Comment by iver56** [comment]
> 
> 
> Maybe I'm missing something. How would you handle the deps specification between
> 
>     * conda env file
> 
>     * requirements.txt
> 
>     * setup.py
>       In particular, the versions are currently fixed in the conda env file (`==`), is this necessary? Would you also fix it in the setup.py?

* setup.py should not be very strict when it comes to ranges of versions.
* Currently, environment.yml shows one of many possible combinations of package versions that should work. I think it makes sense to pin packages at least some packages for development and testing.
* I've never worked on a project that had both environment.yml and requirements.txt. The reason why I prefer conda is that it's the only way I've been able to install pytorch with CUDA support on Windows.
* In other packages, I've been handling dependencies in requirements.txt and setup.py separately. E.g. `audiomentations` has additional dependencies in requirements.txt and more strict version specs in requirements.txt compared to setup.py

**Comment by iver56** [comment]
> 
> 
> My guess is that setting up conda will take quite some time.
> Also, how does it behave to install cudatoolkit on a CPU-only machine?

I see. How about instead making a Dockerfile for running tests in Github Actions? That seems more feasible, doen't it?

Re cudatoolkit on a CPU-only machine: I've been doing `RUN sed -i.bak "s|- cudatoolkit=10.1|- cpuonly|" environment.yml` to deal with this

**Comment by mpariente** [comment]
IDK, will we need ffmpeg during testing?

**Comment by iver56** [comment]
> 
> 
> IDK, will we need ffmpeg during testing?

It depends. Do we need to load special audio file formats, like ogg/opus in testing?

**Comment by mpariente** [comment]
No, I don't think we'll ned any of this. That's why I don't think it's worth it to setup conda for CI. 
I'll add a req file in ./github/workflows with all what's needed for testing and we'll see how we handle deps later.

**Comment by mpariente** [comment]
Ready to merge IMO

**Pull request merged**